### PR TITLE
Collect logs during testing

### PR DESCRIPTION
**Pull request by atodorov**
by adding links to $TEST_ATTACHEMENTS directory and the files will
be uploaded automatically

### CODE REVIEW

**Comment by atodorov** [comment]
@martinpitt  can you point me to an example how to use $TEST_ATTACHMENTS ? I tried similar change in `test/run` before and it could upload some files but not all of them (b/c they were not created)

**Comment by atodorov** [comment]
@bcl - the way beakerlib works we don't exist on failure (even though we hack beakerlib and try to) and the compose usually gets removed at the end which in turn removes logs in `/var/lib/lorax/composer/results`. Should we add an option to remove images but retail their logs ?

**Comment by bcl** [comment]
> Should we add an option to remove images but retail their logs ?
Does that include the logs from trying to boot the image? Those could be useful in debugging why a build succeeds but the boot/ssh/etc. tests fail.

**Comment by larskarlitski** [comment]
>  Should we add an option to remove images but retail their logs ?

I would download the logs after the compose has run and save them in `TEST_ATTACHMENTS` before deleting the compose. `composer-cli` can download all logs as a `tar.gz`.

**Comment by atodorov** [comment]
> I would download the logs after the compose has run

this is done inside the VM/test script

> and save them in TEST_ATTACHMENTS 

this is done outside the VM (test runner essentially)

> deleting the compose. 

this is again inside the VM/test script


If I understand how things work correctly the above procedure is certainly doable but will require changes to every test script to save the logs and then copy them outside the VM so they can be uploaded.

Let me know if my understanding is correct and if you want a separate PR for it or add the changes here.

**Comment by atodorov** [comment]
@larskarlitski, @jikortus, I was able to easily collect the logs from building a new compose however qemu.log always has a size of 0. Can you take a look at this PR and comment on its approach or obvious things I am missing.

File: test/composertest.py [hunk 1]
```diff
@@ -54,6 +54,9 @@ def setUp(self):
         self.assertEqual(r.returncode, 0)
 
     def tearDown(self):
+        # copy logs out of the VM
+        self.machine.download_dir('/var/log/tests', os.environ.get('TEST_ATTACHMENTS', '/tmp'))
```
[the remark this answers is not in the document]
Comment (atodorov) [reply test/composertest.py]: Updated. I will merge after tests report PASS once again.

**Pull request merged**